app/models/course/assessment.rb

```diff
@@ -135,10 +135,18 @@ def initialize_duplicate(duplicator, other)
     copy_attributes(other, duplicator.time_shift)
     self.folder = duplicator.duplicate(other.folder)
     self.questions = duplicator.duplicate(other.questions.map(&:actable)).compact.map(&:acting_as)
-    self.assessment_conditions = duplicator.duplicate(other.assessment_conditions)
-    # Like achievement conditions, duplicate the actable object directly and let the acting_as
-    # gem create the Condition object.
-    self.conditions = duplicator.duplicate(other.conditions.map(&:actable)).map(&:acting_as)
+    if duplicator.mode == :course
+      self.assessment_conditions = duplicator.duplicate(other.assessment_conditions)
+      # Like achievement conditions, duplicate the actable object directly and let the acting_as
+      # gem create the Condition object.
+      self.conditions = duplicator.duplicate(other.conditions.map(&:actable)).map(&:acting_as)
+    elsif duplicator.mode == :object
+      target_category = duplicator.options[:target_course].assessment_categories.first
```

Comment:
The issue is that during duplication, the `initialize_duplicate` function will only have access to the duplicator, not the duplication service.

Unless you are suggesting that we allow the duplicator to access the service? ie. `duplicator.service.target_course`?